presto-main/src/main/java/com/facebook/presto/operator/TableWriterOperator.java

```diff
@@ -124,9 +135,24 @@ public Operator createOperator(DriverContext driverContext)
 
         private ConnectorPageSink createPageSink()
         {
+            ConnectorId connectorId = getConnectorId(target);
+            TaskMetadataContext metadataContext;
+            Optional<ConnectorMetadataUpdater> metadataUpdater = metadataUpdaterManager.getMetadataUpdater(connectorId);
+            if (metadataUpdater.isPresent()) {
+                if (!taskContext.getTaskMetadataContext().isPresent()) {
+                    metadataContext = new TaskMetadataContext(connectorId);
+                    taskContext.setTaskMetadataContext(metadataContext);
+                }
+
+                metadataContext = taskContext.getTaskMetadataContext().get();
+                metadataContext.addMetadataUpdater(metadataUpdater.get());
```

Comment:
Not a big fan of having setter and getter here. If we have to have something mutable, I would say make metadataContext in taskContext immutable and connectorId in metadataContext mutable. That will make this cleaner.

Reply:
Done. Made the `ConnectorId` in `TaskMetadataContext` mutable